### ISSUE TITLE
bug: Non-admins can not create repos for organizations

### ISSUE THREAD

**Issue by MGChecker** [go-gitea/gitea#16575]
If a non-admin tries to create a repository inside an organization they own, they get an error their repository limit would be reached. In my configuration the default repository limit for users is 0.

Specifically, the following error is shown:
form.reach_limit_of_creation%!(EXTRA int=0)

Since organizations are similar to users, that probably means that they have an internal repository limit as well which is incorrectly applied.

- Gitea version (or commit ref): 1.14.5
- Operating system: Debian Buster
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
- Log gist:

**Comment by noerw** [comment]
A discussion about this was started here https://github.com/go-gitea/gitea/pull/15924#issuecomment-846193189, but we had no concluding decision.

It's unclear what the best behaviour for this is, but I agree that it can be confusing.
Changing this is a major breaking change, unless we can come up with a mechanism that easily transitions instances that depend on this behaviour as it currently is.

---

A quick proposal:  introduce a new instance-wide or per-user setting `ORG_REPO_LIMIT`, that
- defaults to `MAX_CREATION_LIMIT` when unset,  
- ignores MAX_CREATION_LIMIT for org repos, when set to -1 (or some better encoding..)
- applies a different limit on org repos for values >= 0

Gitea has enough config options already, so there should be enough people wanting this feature to justify adding this complexity..

**Comment by MGChecker** [comment]
I think this is unnecessarily complicated.

In contrast, I would propose not using the limit of the organization to check whether a repository may be created, but the limit of the user creating the repository at this moment. This is basically possible already by creating the repo in your own user namespace and then transfering it to the organization.

**Comment by noerw** [comment]
> I would propose not using the limit of the organization to check whether a repository may be created, but the limit of the user creating the repository at this moment. 

@MGChecker this is how it works currently. One of us is misunderstanding something here, not sure who :sweat_smile:

**Comment by MGChecker** [comment]
I agree that this is how it should work, but it does not in my setup.

Even though the user repository limit is larger than 0, they may still not create a repository inside the organization. The form is shown, but on submit the error message mentioned above is shown.

**Comment by MGChecker** [comment]
I think this is a misunderstanding and this is actually a bug.

**Comment by brianrobt** [comment]
> Even though the user repository limit is larger than 0, they may still not create a repository inside the organization. The form is shown, but on submit the error message mentioned above is shown.

@MGChecker, maybe I am misunderstanding, but in your original post, isn't the default user repo limit in your setup 0?

I just tried to replicate this in the latest version (v1.23.4) and couldn't replicate the issue:

- User repo limit set to 0
- Organization repo limit set to -1

Result was that the user (organization owner) was not allowed to create an organization repo, which I would expect.

I could possibly see a use case where you'd want a user to be able to create repos in an organization but not in any other public space on the server, but I'm not sure if that's the issue being pointed out here.

This issue is 4 years old, though.  Maybe I'm completely off base.

**Comment by wxiaoguang** [comment]
It should have been fixed in 1.23.7
